js/main.js

```diff
@@ -1,52 +1,94 @@
-// !!!  ДЗ 17. Функція із замиканням
+// !!  ДЗ 20. Створюємо сутності
 
-// ??? Написати функцію, яка приймає 1 параметр. з тим, що передали перший раз і т. д. Все це із замиканнями, наприклад: sum(3) = 3 sum(5) = 8 sum(20) = 28
+// Створити сутність людини:
+// імʼя
+// вік
+// Метод виведення даних
+// Створити сутність автомобіля:
+// Характеристики автомобіля окремими властивостями
+// Методи:
+// Виведення на екран даних про цей автомобіль
+// Привласнення цього автомобіля власнику (записати в автомобіль обʼєкт власника)
 
-// !!!  Рішення
+// !! Рішення
 
-// !!!  варіант 1
+// Що потрібно зробити
 
-// const sum = (function () {
-//     let result = 0; 
-//     return function(number) {
-//        result += number; 
-//        return result; 
-//     } })();
+// 1. Створити клас Людина
+//     властивості 
+//         імʼя
+//         вік
+//     методи
+//         конструктор, який приймає два параметри: імʼя та вік
+//         метод, який виводить у консоль інформацію про людину
 
-// console.log(sum(3));
-// console.log(sum(5));
-// console.log(sum(20));
+class Person {
+  name;
+  age;
 
-// чи так, якщо привести функцію до const і далі вже її виводити
-// як краще???
+  constructor (name, age) {
+    this.name = name;
+    this.age = age;
+  }
+  getInfo() {
+    console.log(`\nВласник:\n\nІм'я = ${this.name}\nВік = ${this.age}`);
+}}
 
-// !!!  варіант 2
+// 2. клас Автомобіль
+//     властивості 
+//         марка, модель, рік виписку, номерний знак (або на Ваш розсуд)
+//         власник
+//     методи 
+//         конструктор, який приймає чотитри параметри (тобто, всі окрім власника): марка, модель, рік виписку, номерний знак 
+//         присвоїти власника - метод повинен приймати екземпляр класу Людина, та зберігати екземпляр класу Людина у відповідному полі, якщо вік більше 18, інакше виводити у консоль відповідне повідомлення
+//         метод, який виводить у консоль інформацію про автомобіль та викликає метод виводу інформації класу Людина для виведення інформації про власника
 
-// const sumFunc = function () {
-//     let result = 0; 
-//     return function(number) {
-//        result += number; 
-//        return result; 
-//     } };
+class Car {
+  brand;
+  model;
+  carYear;
+  carNumber;
+  owner;
 
-// const sum = sumFunc();
+  constructor (brand, model, carYear, carNumber) {
+    this.brand = brand;
+    this.model = model;
+    this.carYear = carYear;
+    this.carNumber = carNumber;
+  }
 
-// console.log(sum(3));
-// console.log(sum(5));
-// console.log(sum(20)); 
+  setOwner (owner){
+    if (owner.age < 18) {
+      return console.log ('Вік меньше 18 років')
+    }
+    this.owner = owner;
+  }
 
-// !!!  варіант 3
+  getCarInfo (){
```

Comment:
```suggestion
   logCarInfo (){
```
метод нічого не повертає